.github/workflows/main.yml

```diff
@@ -0,0 +1,134 @@
+name: GH
+
+on:
+  pull_request:
+  push:
+    branches: master
+  release:
+    types: [released, prereleased]
+  workflow_dispatch:  # allows running workflow manually from the Actions tab
+
+jobs:
+
+  build-sdist:
+    runs-on: ubuntu-latest
+
+    env:
+      PIP_DISABLE_PIP_VERSION_CHECK: 1
+
+    steps:
+    - uses: actions/checkout@v4
+      with:
+        fetch-depth: 0
+
+    - name: Set up Python
+      uses: actions/setup-python@v5
+      with:
+        python-version: '3.x'
+
+    - run: sudo apt-get install -y libmagic1
+
+    - name: Build source distribution
+      run: |
+        pip install -U setuptools wheel pip
+        python setup.py sdist
+
+    - uses: actions/upload-artifact@v4
+      with:
+        name: dist
+        path: dist/*.tar.*
+
+
+  build-wheels-matrix:
+    runs-on: ubuntu-latest
+    outputs:
+      include: ${{ steps.set-matrix.outputs.include }}
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/setup-python@v5
+        with:
+          python-version: '3.x'
+      - run: pip install cibuildwheel==2.17.0  # sync version with pypa/cibuildwheel below
+      - id: set-matrix
+        env:
+          CIBW_PROJECT_REQUIRES_PYTHON: '==3.12.*'
+        run: |
+          MATRIX_INCLUDE=$(
+            {
+              cibuildwheel --print-build-identifiers --platform linux --arch all | grep cp | jq -nRc '{"only": inputs, "os": "ubuntu-latest"}' \
+              && cibuildwheel --print-build-identifiers --platform macos --arch x86_64 | grep cp |  jq -nRc '{"only": inputs, "os": "macos-13"}' \
+              && cibuildwheel --print-build-identifiers --platform macos --arch arm64 | grep cp |  jq -nRc '{"only": inputs, "os": "macos-14"}' \
+              && cibuildwheel --print-build-identifiers --platform windows --arch x86,AMD64 | grep cp |  jq -nRc '{"only": inputs, "os": "windows-latest"}'
+            } | jq -sc
+          )
+          echo "include=$MATRIX_INCLUDE" >> $GITHUB_OUTPUT
+
+
+  build-wheels:
+    name: Build ${{ matrix.only }}
+    needs: build-wheels-matrix
+    runs-on: ${{ matrix.os }}
+
+    strategy:
+      fail-fast: false
+      matrix:
+        include: ${{ fromJson(needs.build-wheels-matrix.outputs.include) }}
+
+    steps:
+    - uses: actions/checkout@v4
+      with:
+        fetch-depth: 0
+
+    - name: Set up QEMU
+      if: runner.os == 'Linux'
+      uses: docker/setup-qemu-action@v3
+
+    - uses: pypa/cibuildwheel@v2.17.0  # sync version with pip install cibuildwheel above
+      timeout-minutes: 10
+      with:
+        only: ${{ matrix.only }}
+      env:
+        CIBW_BUILD_VERBOSITY: 1
+        CIBW_BEFORE_BUILD: bash "add_libmagic.sh"
+        CIBW_TEST_COMMAND: python -c "import magic; assert magic.Magic(mime=True).from_buffer(b'\x00\x00\x00\x1cftypiso5\x00\x00\x00\x01isomiso5hlsf\x00\x00') == 'video/mp4'"
```

Comment:
a simple smoke test: this is an HLS MP4 video, only recognised in recent versions of libmagic.

Reply:
this also runs (without PyPI upload) on pull requests, so hereby adding CI

Reply:
fwiw, I tried switching from this smoketest to `pytest` (https://github.com/ddelange/python-magic/commit/84667ee7d3d6bbc6bcf2400105ed322de22417fa), but got quite many errors on Windows:

<details>
<summary>Logs</summary>

https://github.com/ddelange/python-magic/actions/runs/8966931092/job/24623426057#step:4:1293

```pytb
_________________________ MagicTest.test_descriptions _________________________
  self = <test.python_magic_test.MagicTest testMethod=test_descriptions>
      def test_descriptions(self):
          m = magic.Magic()
          os.environ["TZ"] = "UTC"  # To get last modified date of test.gz in UTC
          try:
  >           self.assert_values(
                  m,
                  {
                      "elf-NetBSD-x86_64-echo": (
                          "ELF 64-bit LSB shared object, x86-64, version 1 (SYSV)",
                          "ELF 64-bit LSB pie executable, x86-64, version 1 (SYSV), dynamically linked, interpreter /libexec/ld.elf_so, for NetBSD 8.0, not stripped",
                      ),
                      "magic._pyc_": "python 2.4 byte-compiled",
                      "test.pdf": (
                          "PDF document, version 1.2",
                          "PDF document, version 1.2, 2 pages",
                          "PDF document, version 1.2, 2 page(s)",
                      ),
                      "test.gz": (
                          'gzip compressed data, was "test", from Unix, last '
                          "modified: Sun Jun 29 01:32:52 2008",
                          'gzip compressed data, was "test", last modified'
                          ": Sun Jun 29 01:32:52 2008, from Unix",
                          'gzip compressed data, was "test", last modified'
                          ": Sun Jun 29 01:32:52 2008, from Unix, original size 15",
                          'gzip compressed data, was "test", '
                          "last modified: Sun Jun 29 01:32:52 2008, "
                          "from Unix, original size modulo 2^32 15",
                          'gzip compressed data, was "test", last modified'
                          ": Sun Jun 29 01:32:52 2008, from Unix, truncated",
                      ),
                      "text.txt": "ASCII text",
                      "test.snappy.parquet": ("Apache Parquet", "Par archive data"),
                      "test.json": "JSON text data",
                  },
                  buf_equals_file=False,
              )
  D:\a\python-magic\python-magic\test\python_magic_test.py:131: 
  _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
  D:\a\python-magic\python-magic\test\python_magic_test.py:54: in assert_values
      self.assertIn(value, expected_value)
  E   AssertionError: 'ASCII text, with CRLF line terminators' not found in ('ASCII text',)
  _____________________ MagicTest.test_descriptions_no_json _____________________
  self = <test.python_magic_test.MagicTest testMethod=test_descriptions_no_json>
      def test_descriptions_no_json(self):
          m = magic.Magic(check_json=False)
          os.environ["TZ"] = "UTC"  # To get last modified date of test.gz in UTC
          try:
  >           self.assert_values(
                  m,
                  {
                      "elf-NetBSD-x86_64-echo": (
                          "ELF 64-bit LSB shared object, x86-64, version 1 (SYSV)",
                          "ELF 64-bit LSB pie executable, x86-64, version 1 (SYSV), dynamically linked, interpreter /libexec/ld.elf_so, for NetBSD 8.0, not stripped",
                      ),
                      "magic._pyc_": "python 2.4 byte-compiled",
                      "test.pdf": (
                          "PDF document, version 1.2",
                          "PDF document, version 1.2, 2 pages",
                          "PDF document, version 1.2, 2 page(s)",
                      ),
                      "test.gz": (
                          'gzip compressed data, was "test", from Unix, last '
                          "modified: Sun Jun 29 01:32:52 2008",
                          'gzip compressed data, was "test", last modified'
                          ": Sun Jun 29 01:32:52 2008, from Unix",
                          'gzip compressed data, was "test", last modified'
                          ": Sun Jun 29 01:32:52 2008, from Unix, original size 15",
                          'gzip compressed data, was "test", '
                          "last modified: Sun Jun 29 01:32:52 2008, "
                          "from Unix, original size modulo 2^32 15",
                          'gzip compressed data, was "test", last modified'
                          ": Sun Jun 29 01:32:52 2008, from Unix, truncated",
                      ),
                      "text.txt": "ASCII text",
                      "test.snappy.parquet": ("Apache Parquet", "Par archive data"),
                  },
                  buf_equals_file=False,
              )
  D:\a\python-magic\python-magic\test\python_magic_test.py:209: 
  _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
  D:\a\python-magic\python-magic\test\python_magic_test.py:54: in assert_values
      self.assertIn(value, expected_value)
  E   AssertionError: 'ASCII text, with CRLF line terminators' not found in ('ASCII text',)
  _____________________ MagicTest.test_descriptions_no_soft _____________________
  self = <test.python_magic_test.MagicTest testMethod=test_descriptions_no_soft>
      def test_descriptions_no_soft(self):
          m = magic.Magic(check_soft=False)
  >       self.assert_values(
              m,
              {
                  "elf-NetBSD-x86_64-echo": (
                      "data",
                      "ELF 64-bit LSB pie executable, x86-64, version 1 (SYSV), dynamically linked, interpreter /libexec/ld.elf_so, for NetBSD 8.0, not stripped",
                  ),
                  "magic._pyc_": "data",
                  "test.pdf": "ASCII text",
                  "test.gz": "data",
                  "text.txt": "ASCII text",
                  "test.snappy.parquet": "data",
                  "test.json": "JSON text data",
              },
              buf_equals_file=False,
          )
  D:\a\python-magic\python-magic\test\python_magic_test.py:168: 
  _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
  D:\a\python-magic\python-magic\test\python_magic_test.py:54: in assert_values
      self.assertIn(value, expected_value)
  E   AssertionError: 'ASCII text, with CRLF line terminators' not found in ('ASCII text',)
  __________________________ MagicTest.test_mime_types __________________________
  self = <test.python_magic_test.MagicTest testMethod=test_mime_types>
      def test_mime_types(self):
          dest = os.path.join(MagicTest.TESTDATA_DIR, b"\xce\xbb".decode("utf-8"))
          shutil.copyfile(os.path.join(MagicTest.TESTDATA_DIR, "lambda"), dest)
          try:
              m = magic.Magic(mime=True)
  >           self.assert_values(
                  m,
                  {
                      "elf-NetBSD-x86_64-echo": (
                          "application/x-pie-executable",
                          "application/x-sharedlib",
                      ),
                      "magic._pyc_": (
                          "application/octet-stream",
                          "text/x-bytecode.python",
                          "application/x-bytecode.python",
                      ),
                      "test.pdf": "application/pdf",
                      "test.gz": ("application/gzip", "application/x-gzip"),
                      "test.snappy.parquet": "application/octet-stream",
                      "text.txt": "text/plain",
                      b"\xce\xbb".decode("utf-8"): "text/plain",
                      b"\xce\xbb": "text/plain",
                      "test.json": "application/json",
                  },
                  buf_equals_file=False,
              )
  D:\a\python-magic\python-magic\test\python_magic_test.py:102: 
  _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
  D:\a\python-magic\python-magic\test\python_magic_test.py:54: in assert_values
      self.assertIn(value, expected_value)
  E   AssertionError: "cannot open `D:\\\\a\\\\python-magic\\\\python-magic\\\\test\\\\testdata\\\\\u03bb' (No such file or directory)" not found in ('text/plain',)
  ___________________________ MagicTest.test_symlink ____________________________
  self = <test.python_magic_test.MagicTest testMethod=test_symlink>
      def test_symlink(self):
          # TODO: 3.0
          if not hasattr(tempfile, "TemporaryDirectory"):
              return
      
          with tempfile.TemporaryDirectory() as tmp:
              tmp_link = os.path.join(tmp, "test_link")
              tmp_broken = os.path.join(tmp, "nonexistent")
      
              os.symlink(
                  os.path.join(self.TESTDATA_DIR, "test.pdf"),
                  tmp_link,
              )
      
              os.symlink("/nonexistent", tmp_broken)
      
              m = magic.Magic()
              m_follow = magic.Magic(follow_symlinks=True)
  >           self.assertTrue(m.from_file(tmp_link).startswith("symbolic link to "))
  E           AssertionError: False is not true
  D:\a\python-magic\python-magic\test\python_magic_test.py:373: AssertionError
```

</details>